### DISCUSSION TITLE
Cache hit occurred on the primary key

### DISCUSSION THREAD

**Discussion by shawngraham** [CollectionBuilder/collectionbuilder.github.io#43] [Q&A]
Hi,

I have collectionbuilder csv template working, everything has been great up till now. I'm using github actions to deploy, and suddenly the build is failing, with this in the logs marked as the fail:

```
save bundler cache
  Cache hit occurred on the primary key Linux-gems-7f08d513391365250a119724af1a5a998b72174c71bacd296366975988cd9005, not saving cache.
``` 

Any ideas?

**Comment by dcnb** [comment]
Hi @shawngraham  -- it looks like your repo is deploying again. Did you figure it out? 

I have not seen that error before. @evanwill uses GitHub actions more though. He may have some idea.

**Reply by shawngraham** [reply]
Ultimately, I discovered that it was caused by the encoding on the csv being changed from utf-8 to something else!

**Reply by evanwill** [reply]
@shawngraham and @dcnb the "Cache hit" message is a red herring showing up in the github actions message (just means that it isn't saving the cache because it already exists)--> the Jekyll build error actually happens above that point but isn't highlighted by the interface for some reason... 
On the GitHub Actions output, e.g. https://github.com/shawngraham/test-collectionbuilder/runs/4846318749?check_suite_focus=true
click the "jekyll build" step under "Run limjh16/jekyll-action-ts@v2" to see the build error output-- and as you discovered it is CSV encoding issue.

The *not* UTF-8 or corrupted encoding is a common error that can be hard to debug... it is a Ruby + Jekyll issue (not CollectionBuilder specific!). Most commonly happens when trying to use Excel (Excel really can not create properly formatted utf-8 CSV, at best creates “UTF-8 with BOM” which breaks Jekyll), but also seems to come up occasionally when pasting between files (on Mac??) which I haven't figured out exactly. 

@shawngraham did you figure out what was doing it to your CSV?

**Reply by shawngraham** [reply]
Yeah, it was totally excel. Excel was inserting line breaks in the middle of data! And then there would be empty rows... God I hate excel.
